web/src/main/java/org/apache/shiro/web/filter/authz/IpAddressMatcher.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2002-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.shiro.web.filter.authz;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+
+/**
+ * Matches a request based on IP Address or subnet mask matching against the remote
+ * address.
+ * <p>
+ * Both IPv6 and IPv4 addresses are supported, but a matcher which is configured with an
+ * IPv4 address will never match a request which returns an IPv6 address, and vice-versa.
+ *
+ * @see <a href="https://github.com/spring-projects/spring-security/blob/master/web/src/main/java/org/springframework/security/web/util/matcher/IpAddressMatcher.java">Original Spring Security version</a>
+ * @since 1.5
```

Comment:
This needs to be set to the next version

Reply:
@bdemers I was thinking to release it in the 1.5.4 too. Thoughts?

Reply:
I have no objection to another 1.x release. Though my pref for this change would be a 1.6 (as it is a new feature)

We can also, just mark this as some sort of placeholder like `NEXT_VERSION` and then update it before the release (if we are undecided on the exact version)

Reply:
@bdemers ok as it's a new feature. I would prefer to move on 2.0 to keep focus on it.